lib/calculator/lexer.rb

```diff
@@ -178,6 +178,48 @@ def do_right_parenthesis
     # literals at this point.
     class DefaultState < State
       include Singleton
+
+      def do_end_of_input
+        [:accept]
+      end
+      def do_whitespace
+        [self]
+      end
+      def do_full_stop
+        [self]
+      end
+      def do_plus
+        [:self, AddOpToken.instance]
+      end
+      def do_asterisk
+        [:self, MultiplyOpToken.instance]
+      end
+      def do_slash
+        [:self, DivideOpToken.instance]
+      end
+      def do_caret
+        [:self, ExponentOpToken.instance]
+      end
+      def do_left_parenthesis
+        [:self, LeftParenthesisToken.instance]
+      end
+      def do_right_parenthesis
+        [:self, RightParenthesisToken.instance]
+      end
+    end
+    class IntegerPartState < State
+      include Singleton
+
+      def do_digit
+        [self]
+      end
+    end
+    class HyphenState < State
```

Comment:
Neither this state or the integer part state are ever invoked from the default state.